relay-server/src/actors/project_local.rs

```diff
@@ -1,81 +1,108 @@
 use std::collections::HashMap;
 use std::ffi::OsStr;
-use std::fs;
-use std::io;
 use std::path::Path;
 use std::sync::Arc;
-use std::thread;
-
-use actix::prelude::*;
-use futures01::{sync::oneshot, Future};
 
 use relay_common::{ProjectId, ProjectKey};
 use relay_config::Config;
 use relay_log::LogError;
+use relay_system::AsyncResponse;
+use relay_system::FromMessage;
+use relay_system::Interface;
+use relay_system::Sender;
+use relay_system::Service;
 
 use crate::actors::project::ProjectState;
 use crate::actors::project_cache::FetchOptionalProjectState;
 
+/// Service interface of the local project source.
+#[derive(Debug)]
+pub enum LocalProjectSource {
+    /// Fetches the project state for a given project key. Returns `None` if the project state is not available locally.
+    FetchOptionalProjectState(FetchOptionalProjectState, Sender<Option<Arc<ProjectState>>>),
```

Comment:
I considered it, but I wasn't sure how to name these types then. Would I call the `Interface` `FetchOptionalProjectState`, and rename `LocalProjectSourceService` back to `LocalProjectSource`?

Reply:
You can keep the interface on `LocalProjectSource` since you need the sender:

```rust
pub struct LocalProjectSource(FetchOptionalProjectState, Sender<Option<Arc<ProjectState>>>);

impl Interface for LocalProjectSource {}
```